mllib/src/test/scala/org/apache/spark/ml/feature/QuantileDiscretizerSuite.scala

```diff
@@ -91,6 +44,46 @@ class QuantileDiscretizerSuite
       "Observed number of buckets does not equal expected number of buckets.")
   }
 
+  test("Test transform method on unseen data") {
+    val sqlCtx = SQLContext.getOrCreate(sc)
+    import sqlCtx.implicits._
+
+    val trainDF = sc.parallelize(1.0 to 100.0 by 1.0).map(Tuple1.apply).toDF("input")
+    val testDF = sc.parallelize(-10.0 to 110.0 by 1.0).map(Tuple1.apply).toDF("input")
+    val discretizer = new QuantileDiscretizer()
+      .setInputCol("input")
+      .setOutputCol("result")
+      .setNumBuckets(5)
+
+    val result = discretizer.fit(trainDF).transform(testDF)
+    val firstBucketSize = result.filter(result("result") === 0.0).count
+    val lastBucketSize = result.filter(result("result") === 4.0).count
+
+    assert(firstBucketSize === 30L,
+      "Size of first bucket did not equal expected value.")
+    assert(lastBucketSize === 31L,
+      "Size of last bucket did not equal expected value.")
+  }
+
+  test("Test default relativeError and setRelativeError") {
+    val discretizer = new QuantileDiscretizer()
+    val testValues = Array(0.0, 0.005, 0.1)
+
+    discretizer.setNumBuckets(2)
+    assert(discretizer.getRelativeError === 0.01,
+      "Default relativeError value incorrect for <5 buckets.")
+
+    discretizer.setNumBuckets(101)
+    assert(discretizer.getRelativeError === 0.001,
```

Comment:
It doesn't look like `approxQuantile` has a default value for `relativeError`.

https://github.com/apache/spark/blob/13ce10e95401b21fa40ca0bb27ebf9a0bfffe70f/sql/core/src/main/scala/org/apache/spark/sql/DataFrameStatFunctions.scala#L66